### ISSUE TITLE
unable to load python-trained model with libtorch c++ api in Windows

### ISSUE THREAD

**Issue by xinheblue** [pytorch/pytorch#15589]
version: stable 1.0.0
get from the url
https://download.pytorch.org/libtorch/cu90/libtorch-win-shared-with-deps-latest.zip

test code

#include <torch/torch.h>
#include <torch/script.h>

#include<fstream>
#include <iostream>
using namespace std;

int main(int argc, const char* argv[]) {

  at::Tensor tensor = torch::rand({2, 3}).to(at::kCUDA);
  std::cout << tensor << std::endl;

  /**/
  string model_name = "model.pt";
  try {
	  
	  std::ifstream in(model_name, std::ios_base::binary);

	  if (in.fail()) {
		  cout << "failed to open model" << endl;
	  }
	  else {
		  cout << "successed to open model" << endl;
	  }

	  AT_CHECK(!in.fail(), "load: could not open file ", model_name);

	  cout << "parsed checking" << endl;
	   

	  std::shared_ptr<torch::jit::script::Module> module = torch::jit::load(model_name);
	  module->to(at::kCUDA);

	  assert(module != nullptr);
	  std::cout << "ok\n";

	  std::vector<torch::jit::IValue> inputs;
	  inputs.push_back(torch::ones({ 1, 3, 224, 224 }).to(at::kCUDA));

	  // Execute the model and turn its output into a tensor.
	  at::Tensor output = module->forward(inputs).toTensor();

	  std::cout << output.slice(/*dim=*/1, /*start=*/0, /*end=*/5) << '\n';
	 
  }
  catch (exception & err) {
	  cout << err.what() << endl;
  }
   
  

  cout << "finished" << endl;

}

output:

D:\learn\cv\lab\example-app\build6\Debug>example-app.exe  model.pt
 0.6020  0.1421  0.9155
 0.6821  0.7416  0.7934
[ Variable[CUDAFloatType]{2,3} ]
load: could not open file  (load at ..\torch\csrc\jit\import.cpp:250)
(no backtrace available)
finished

the model is the python trained,with the following code
import torch
import torchvision

# An instance of your model.
model = torchvision.models.resnet18()

# An example input you would normally provide to your model's forward() method.
example = torch.rand(1, 3, 224, 224)

# Use torch.jit.trace to generate a torch.jit.ScriptModule via tracing.
traced_script_module = torch.jit.trace(model, example)

# save
traced_script_module.save("model.pt")

it should be in the right position, as the check code shows.

**Comment by peterjc123** [comment]
This is not a bug. For MSVC, `std::string` is not defined the same for the configurations Debug and Release. Since we built libtorch with the __Release__ configuration, if you try to use the __Debug__ configuration when compiling the executable, the `std::string` object could not be passed into the torch library, and hence the error. I tried building it with Release using the following command and it worked fine. 
```cmd
set CMAKE_BUILD_TYPE=Release
msbuild INSTALL.vcxproj /p:Configuration=%CMAKE_BUILD_TYPE%
```

**Comment by xinheblue** [comment]
@peterjc123 I changed the string to ifstream,  it can find the model but still aborts with the following error

**std::_Hash<std::_Umap_traits<std::basic_string<char,std::char_traits<char>,std::allocator<char> >,unsigned __int64,std::_Uhash_compare<std::basic_string<char,std::char_traits<char>,std::allocator<char> >,std::hash<std::basic_string<char,std::char_traits<char>,std::allocator<char> > >,std::equal_to<std::basic_string<char,std::char_traits<char>,std::allocator<char> > > >,std::allocator<std::pair<std::basic_string<char,std::char_traits<char>,std::allocator<char> > const ,unsigned __int64> >,0> >::_Vec_lo**(...) 返回 0x6。

however, when i change the mode from Debug to Release in Visual Studio 2017, still with the string object with input, it works and get the desired error

I test the latest load function in ubuntu 16.04, with gcc 4.9.3,the same code
throws the error
terminte called after throwing an instance of 'std::bad_alloc'
wait(): std::bad_alloc
Aborted(core dumped)
when I comment out the line
#the code
auto output = module->forward(inputs).toTensor();
the programme processed without this error tip

**Comment by xinheblue** [comment]
also, in Release mode, with the ifstream with input, it get the desired output without exception

**Comment by peterjc123** [comment]
@xinheblue Yes, that is expected. We should not mix up the build configurations. So this is not a bug.

**Comment by xinheblue** [comment]
Thanks @peterjc123 , but if one wants to develop in Debug mode with the prebuilt library, the current issue may still exists.

**Comment by peterjc123** [comment]
Then you'll need the debug version. Unluckily, we don't currently provide this.

**Comment by xinheblue** [comment]
thanks, close the issue now
